### PR TITLE
Refactor SARA handling allowing for historic checks

### PR DESCRIPTION
**Pull request by pfcairns**
## Context

During PNI score calculation check for historic SARA calculation which may not be present on latest completed Layer 3 assessment

## Changes in this PR

- Updated the PniService to fetch relationships with completed SARA assessments within 6 weeks of latest completed assessement.
- Added both SARA scores to persisted PNI result
- Enhanced JUnit and Integration tests to cover scenarios with different SARA statuses 
- Renamed test methods for clarity.

**NB. Have left the existing sara field present on IndividualRiskScores as existing prod data may exist with this value populated, and marshalling errors could occur as a result** 

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes UI for this change to work...
  - [ ] ... and they been released to production already

### Post-merge

- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/model/IndividualRiskScores.kt [hunk 1]
```diff
@@ -32,4 +32,13 @@ data class IndividualRiskScores(
 
   @Schema(example = "LOW", description = "")
   @get:JsonProperty("sara") val sara: String? = null,
+
+  @Schema(example = "LOW")
+  @get:JsonProperty("saraRiskOfViolenceTowardsPartner") val saraRiskOfViolenceTowardsPartner: String? = null,
+
+  @Schema(example = "LOW")
+  @get:JsonProperty("saraRiskOfViolenceTowardsOthers") val saraRiskOfViolenceTowardsOthers: String? = null,
```
Comment (anandelangovan1) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/model/IndividualRiskScores.kt:40]: I think we still need to report sara outcome

We need to take whatever is the higher of the 2 saras
Something like:
sara {
  result: "HIGH",
  saraRiskOfViolenceTowardsPartner: "MEDIUM",
  saraRiskOfViolenceTowardsOthers: "HIGH"
}

Comment (pfcairns) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/model/IndividualRiskScores.kt:40]: Good point! Will address now.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/OasysService.kt [hunk 2]
```diff
@@ -208,6 +210,34 @@ class OasysService(
...
+        }
+      }
+    }
+    log.warn("No completed assessment with SARA data found for prison number $prisonNumber")
+    return null
+  }
+
+  fun isWithinSixWeeks(date1: LocalDateTime?, date2: LocalDateTime?): Boolean {
+    val daysBetween = ChronoUnit.DAYS.between(date1, date2)
+    return abs(daysBetween) <= 42 // 6 weeks * 7 days/week = 42 days
```
Comment (anandelangovan1) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/OasysService.kt:238]: same day test?

Comment (pfcairns) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/OasysService.kt:238]: Will add an additional test.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/model/IndividualRiskScores.kt [hunk 3]
```diff
@@ -32,4 +32,13 @@ data class IndividualRiskScores(
 
   @Schema(example = "LOW", description = "")
   @get:JsonProperty("sara") val sara: String? = null,
+
+  @Schema(example = "LOW")
+  @get:JsonProperty("saraRiskOfViolenceTowardsPartner") val saraRiskOfViolenceTowardsPartner: String? = null,
```
Comment (anandelangovan1) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/restapi/model/IndividualRiskScores.kt:37]: As discussed, enums with weights of 0,10..50 to do the comparison to return the sara result.

**Review by anandelangovan1 — APPROVED** [review]

**Pull request merged**